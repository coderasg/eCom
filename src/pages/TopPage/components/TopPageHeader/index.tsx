import React, {  ChangeEvent, useState } from 'react';
import  Input from './../../../../components/Input';

import Menu from './../../../../components/Icons/Menu/index.svg';
import Notification from './../../../../components/Icons/Notification/index.svg';
import Tick from './../../../../components/Icons/Tick/index.svg';
import { $iconDefaultColor } from './../../../../components/Icons/IconColors';

import styles from './topPageHeader.module.scss';

type Props = {
  searchStr: string;
  handleOnSearchChange: (value: string) => void;
};

function TopPageHeader(props: Props) {
  const { handleOnSearchChange, searchStr } = props;
  const [ state, setState] = useState(searchStr);

  const handleOnChange = (event: ChangeEvent<HTMLInputElement>) => {
    setState(event.target.value);
    handleOnSearchChange(event.target.value);
  };

  return (
    <div data-testid="TopPageHeader-wrapper" className={styles.topPageHeader}>
      <section>
        <Menu width={25} height={25} fill={$iconDefaultColor} />
        <Input
          name="searchProduct"
          placeholder="Search product"
          className={styles.input}
          value={state}
          onChange={handleOnChange}
        />
      </section>
      <section className={styles.flexLeft}>
        <Notification width={25} height={25} fill={$iconDefaultColor} />
        <Tick width={25} height={20} fill={$iconDefaultColor} />
      </section>
    </div>
  );
}

export default TopPageHeader;
