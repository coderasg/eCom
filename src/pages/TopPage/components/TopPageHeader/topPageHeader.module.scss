@use "./../../../../variables";

.topPageHeader {
  display: flex;
  width: 100%;
  align-items: center;

  input {
    background-color: variables.$headerInput;
    border-radius: 5px;
    padding: 2px;
    height: 25px;
    width: 65%;
    min-width: 180px;
    border: none;
  }

  section {
    display: flex;
    flex-grow: 1;
    justify-content: space-around;
  }

  @media (min-width: 480px) {
    display: flex;
    justify-content: space-between;

    section {
      flex-grow: 0;
      padding: 0 20px;
    }

    .flexLeft {
      margin-left: auto;
    }
  }
}
