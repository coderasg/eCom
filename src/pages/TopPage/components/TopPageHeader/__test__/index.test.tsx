import React, {ChangeEvent} from 'react';
import renderer from 'react-test-renderer';

import TopPageHeader from '../index';

describe('(Component) TopPageHeader', () => {
  const defaultProps = {
    searchStr: 'search',
    handleOnSearchChange: (value: string) => {
      console.log(value);
    }
  };
  
  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<TopPageHeader {...defaultProps} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
