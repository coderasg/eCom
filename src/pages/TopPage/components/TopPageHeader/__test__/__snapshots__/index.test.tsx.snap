// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) TopPageHeader should pass snapshot test 1`] = `
<div
  data-testid="TopPageHeader-wrapper"
>
  <section>
    <SvgIcon
      fill="#b1b0b5"
      height={25}
      width={25}
    />
    <div
      data-testid="Input-wrapper"
    >
      <input
        data-testid="searchInput"
        name="searchProduct"
        onChange={[Function]}
        placeholder="Search product"
        type="text"
        value="search"
      />
    </div>
  </section>
  <section>
    <SvgIcon
      fill="#b1b0b5"
      height={25}
      width={25}
    />
    <SvgIcon
      fill="#b1b0b5"
      height={20}
      width={25}
    />
  </section>
</div>
`;
